pinot-controller/src/main/java/org/apache/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -222,8 +222,16 @@ private void checkCRC(HttpHeaders headers, String offlineTableName, String segme
 
   private void processNewSegment(SegmentMetadata segmentMetadata, URI finalSegmentLocationURI,
       File currentSegmentLocation, String zkDownloadURI, HttpHeaders headers, String crypter, String tableNameWithType,
-      String segmentName, boolean moveSegmentToFinalLocation)
+      String segmentName, boolean moveSegmentToFinalLocation, boolean enableParallelPushProtection)
       throws Exception {
+    SegmentZKMetadata newSegmentZKMetadata = _pinotHelixResourceManager
+        .constructZkMetadataForNewSegment(tableNameWithType, segmentMetadata, zkDownloadURI, crypter,
+            enableParallelPushProtection);
```

Comment:
Suggest not passing in `enableParallelPushProtection` but check it in this method and call `setSegmentUploadStartTime()` within this method to limit the scope of parallel push protection

Reply:
Done.